src/main/java/com/cloudbees/jenkins/plugins/sshagent/exec/ExecRemoteAgent.java

```diff
@@ -44,8 +44,6 @@
     private static final String AuthSocketVar = "SSH_AUTH_SOCK";
     private static final String AgentPidVar = "SSH_AGENT_PID";
     
-    private final Launcher launcher;
-    
     /**
      * The listener in case we need to report exceptions
```

Comment:
Probably better to pass the `TaskListener` as an argument to these methods too, so we do not need to rely on `Launcher.getListener()`.